### PR TITLE
OpenAPI: Automation Framework Support

### PR DESCRIPTION
**Pull request by psiinon**
Also adds basic statistics support as well.

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Example cmdline success output:
```
Job addOns set updateAddOns = false
Job addOns started
Job addOns finished
Job openapi set apifile = /Users/simon/Documents/ZAP/devs/autoframework/petstore-expanded.json
Job openapi set targeturl = http://localhost:8080/
Job openapi started
Job openapi added 4 URLs
Job openapi finished
Automation plan succeeded!
```
Example cmdline failure output:
```
Job addOns set updateAddOns = false
Job addOns started
Job addOns finished
Job openapi set apifile = /Users/simon/Documents/ZAP/devs/autoframework/petstore-expandedX.json
Job openapi set targeturl = http://localhost:8080/
Job openapi started
Job openapi cannot read file: /Users/simon/Documents/ZAP/devs/autoframework/petstore-expandedX.json
Job openapi finished
Automation plan failures:
	Job openapi cannot read file: /Users/simon/Documents/ZAP/devs/autoframework/petstore-expandedX.json
```

File: addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    parameters:
+      apifile:                         # String: Local file containing the OpenAPI definition, default: null, no definition will be imported
+      apiurl:                          # String: URL containing the OpenAPI definition, default: null, no definition will be imported
+      targeturl:                       # String: URL which overrides the target defined in the definition, default: null, the target will not be overriden
+</pre>
+
+<H2>Statistics</H2>
+The add-on maintains the following statistics:
+<ul>
+<li>openapi.urls.added : The total number of URLs added when importing OpenAPI definitions
```
Comment (kingthorin) [inline addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html:30]:
```suggestion
<li>openapi.urls.added : The total number of URLs added when importing OpenAPI definitions</li>
```

**Review by kingthorin — CHANGES_REQUESTED** [review]
Looks good to me, just one minor thing in the help.

File: addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html (hunk 1 above)
Comment (psiinon) [reply addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html]: Fixed :)

**Review by kingthorin — APPROVED** [review]

File: addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/toc.xml [hunk 2]
```diff
@@ -7,6 +7,7 @@
 	<tocitem text="ZAP User Guide" tocid="toplevelitem">
 		<tocitem text="Add Ons" tocid="addons">
 			<tocitem text="OpenAPI Support" image="openapi-icon" target="openapi"/>
+			<tocitem text="OpenAPI Automation" target="openapi.automation"/>
```
Comment (thc202) [inline addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/toc.xml:10]: Should be under the previous `tocitem`.

File: addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+OpenAPI Automation Framework Support
+</TITLE>
+</HEAD>
+<BODY>
+<H1>OpenAPI Automation Framework Support</H1>
+This add-on supports the Automation Framework.
+<br><br>
+The add-on will add OpenAPI definitions if they are found while spidering but adding them explicitly via a URL or local file is recommended if they are available.
+<br><br>
+The targeturl parameter works in the same way as <a href="openapi.html">'Target URL Format'</a> in the UI.
```
Comment (thc202) [inline addOns/openapi/src/main/javahelp/org/zaproxy/zap/extension/openapi/resources/help/contents/automation.html:15]: The format applies to API as well, reword to not say just "UI"?

**Comment by psiinon** [comment]
Fixed those comments :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**